src/components/panel/panel.js

```diff
@@ -861,20 +837,15 @@ MdPanelRef.prototype.open = function() {
  * closed and animations finish.
  */
 MdPanelRef.prototype.close = function() {
-  // TODO(ErinCoughlan) - Cancel any in-progress actions.
-
   var self = this;
-  var closePromise = this.hide()
-      .then(function() {
-        return self.detach();
-      }, this._$q.reject)
-      .then(function() {
-        // TODO(ErinCoughlan) - Add destroy. This will make the code here
-        // different than just calling this.detach().
-        return self._$q.resolve(self);
-      }, this._$q.reject);
-
-  return closePromise;
+  return this._$q(function(resolve, reject) {
```

Comment:
Careful with the cluttering of rejects. I contend that this is a better structure:

```js
MdPanelRef.prototype.close = function() {
 var self = this; 
 
return this._$q(function(resolve, reject) {
    var done = angular.bind(self,resolve, self),
        detach = angular.bind(self, self.detach);
        
    self.hide()
        .then( detach )
        .then( done )
        .catch( reject );
 });
};
```

> ^ is untested but I believe the logic is valid and more clear.